cmd/create/cluster/shared_vpc.go

```diff
@@ -0,0 +1,96 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/rosa"
+	"github.com/spf13/cobra"
+	errors "github.com/zgalor/weberr"
+)
+
+func isSubnetBelongToSharedVpc(r *rosa.Runtime, accountID string, subnetIDs []string,
+	mapSubnetToOwnerID map[string]string) bool {
+	for _, subnetID := range subnetIDs {
+		ownerID := mapSubnetToOwnerID[subnetID]
+		if ownerID != "" && ownerID != accountID {
+			r.Reporter.Infof(fmt.Sprintf("Subnet with ID '%s' is shared by AWS account '%s', "+
+				"Defaulting to installing a cluster into a shared VPC.", subnetID, ownerID))
+			return true
+		}
+	}
+
+	return false
+}
+
+func getPrivateHostedZoneID(cmd *cobra.Command, privateHostedZoneID string) (string, error) {
+	res, err := interactive.GetString(interactive.Input{
+		Question: "Private hosted zone ID",
+		Help:     cmd.Flags().Lookup("private-hosted-zone-id").Usage,
+		Default:  privateHostedZoneID,
+		Required: true,
+	})
+	if err != nil {
+		return "", errors.Errorf("Expected a valid value for 'private-hosted-zone-id': %s", err)
+	}
+
+	return res, nil
+}
+
+func getSharedVpcRoleArn(cmd *cobra.Command, sharedVpcRoleArn string) (string, error) {
+	res, err := interactive.GetString(interactive.Input{
+		Question: "Shared VPC role ARN",
+		Help:     cmd.Flags().Lookup("shared-vpc-role-arn").Usage,
+		Default:  sharedVpcRoleArn,
+		Required: true,
+		Validators: []interactive.Validator{
+			aws.ARNValidator,
+		},
+	})
+	if err != nil {
+		return "", errors.Errorf("Expected a valid value for 'shared-vpc-role-arn': %s", err)
+	}
+
+	return res, nil
+}
+
+func getBaseDomain(r *rosa.Runtime, cmd *cobra.Command, baseDomain string) (string, error) {
+	dnsDomains, err := getAvailableBaseDomains(r)
+	if err != nil {
+		return "", err
+	}
+
+	res, err := interactive.GetOption(interactive.Input{
+		Question: "Base Domain",
+		Help:     cmd.Flags().Lookup("base-domain").Usage,
+		Default:  baseDomain,
+		Required: true,
+		Options:  dnsDomains,
+	})
+	if err != nil {
+		return "", errors.Errorf("Expected a valid value for 'base-domain': %s", err)
+	}
+
+	return res, nil
+}
+
+func getAvailableBaseDomains(r *rosa.Runtime) ([]string, error) {
+	organizationID, _, err := r.OCMClient.GetCurrentOrganization()
+	if err != nil {
+		return nil, errors.Errorf("Failed to get current OCM organization ID: %s", err)
+	}
+
+	dnsDomains, err := r.OCMClient.ListDNSDomains(
+		fmt.Sprintf("user_defined='true' and cluster.id='' and organization.id='%s'", organizationID))
+	if err != nil {
+		return nil, errors.Errorf("Failed to list DNS domains: %s", err)
+	}
+
+	var dnsDomainsIDs []string
+	for _, dnsDomain := range dnsDomains {
+		dnsDomainsIDs = append(dnsDomainsIDs, dnsDomain.ID())
```

Comment:
do we need to filter out the dnsdomain which already has a cluster linked here?

Reply:
We do filter out using this query `"user_defined='true' and cluster.id='' and organization.id='%s'"`
Do you think it's better to display all domains? and get the error from the backend?

Reply:
that makes sense. I ignored the cluster.id='' before